components/image-builder-bob/README.md

```diff
@@ -1,3 +1,16 @@
+# Introduction to image-builder-bob
+
+## Overview
+
+Bob is a CLI responsible for building and pushing workspace images during workspace startup.
+
+For each image build, a headless workspace gets created in the meta cluster by `image-builder-mk3` ([#7845](https://github.com/gitpod-io/gitpod/issues/7845) will move the headless workspace and image-builder to the workspace cluster), in this headless workspace runs:
+- `bob proxy`, which gets started by workspacekit in ring1, and receives credentials for pushing images to a docker registry. It proxies and authenticates the image pushes from `bob build`.
+- `bob build` as a workspace task, which builds the
+  - **base layer**, if a custom Dockerfile is specified in `.gitpod.yaml`. If this base image has already been built for the workspace, this step is skipped, and the reference of the previously built image is used instead to build the workspace image next.
+  - **workspace image**, which builds an image from the base layer, where the base layer is either a previously built custom Dockerfile or a public image.
+  These images get pushed over `localhost` to `bob proxy`, as `bob build` does not receive the credentials to push to private registries.
```

Comment:
:100: It helps me to understand image-builder-bob. I wanted this README when I first developed image-builder.

What about putting a statement that this image does not include the supervisor, etc.? I think this is often misunderstood as the role of the registry-facade.

Reply:
Thanks @utam0k great suggestion, updated and will also put it in the architecture docs.

Reply:
> 💯 It helps me to understand image-builder-bob. I wanted this README when I first developed image-builder.

+1